src/Symfony/Component/Security/Http/EventListener/IsCsrfTokenValidAttributeListener.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ControllerArgumentsEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Security\Core\Exception\InvalidCsrfTokenException;
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+use Symfony\Component\Security\Http\Attribute\IsCsrfTokenValid;
+
+/**
+ * Handles the IsCsrfTokenValid attribute on controllers.
+ */
+class IsCsrfTokenValidAttributeListener implements EventSubscriberInterface
```

Comment:
Do we want to make it final?

Reply:
I think so 👍 The container wouldn't be able to properly handle any potential extra capacity added to a child attribute, hence extending autoconfigured attributes isn't supported basically.